src/main/java/de/hysky/skyblocker/skyblock/dungeon/LividColor.java

```diff
@@ -44,37 +46,51 @@ public class LividColor {
-            if ((config.enableLividColorText || config.enableLividColorTitle || config.enableLividColorGlow) && message.getString().equals("[BOSS] Livid: I respect you for making it to here, but I'll be your undoing.")) {
-                tenTicks = 8;
-            }
-        });
+        ClientPlayConnectionEvents.JOIN.register((handler, sender, client) -> LividColor.reset());
+        WorldRenderEvents.AFTER_ENTITIES.register(LividColor::update);
     }
 
-    public static void update() {
+    private static void update(WorldRenderContext context) {
+        DungeonsConfig.Livid config = SkyblockerConfigManager.get().dungeons.livid;
+        if (!(config.enableLividColorText || config.enableLividColorTitle || config.enableLividColorGlow || config.enableLividColorBoundingBox)) return;
+
         MinecraftClient client = MinecraftClient.getInstance();
-        if (tenTicks != 0) {
-            DungeonsConfig.Livid config = SkyblockerConfigManager.get().dungeons.livid;
-            if ((config.enableLividColorText || config.enableLividColorTitle || config.enableLividColorGlow) && Utils.isInDungeons() && client.world != null) {
-                if (tenTicks == 1) {
-                    onLividColorFound(client, Blocks.RED_WOOL);
-                    return;
-                }
-                Block color = client.world.getBlockState(new BlockPos(5, 110, 42)).getBlock();
-                if (WOOL_TO_FORMATTING.containsKey(color) && !color.equals(Blocks.RED_WOOL)) {
-                    onLividColorFound(client, color);
-                    return;
-                }
-                tenTicks--;
-            } else {
-                tenTicks = 0;
+
+        if (!(Utils.isInDungeons() && DungeonManager.isInBoss() && client.player != null && client.world != null)) return;
+
+        Block currentColor = client.world.getBlockState(new BlockPos(5, 110, 42)).getBlock();
+        if (!(WOOL_TO_FORMATTING.containsKey(currentColor) && !currentColor.equals(lastColor))) return;
+
+        if (!isInitialized && client.player.hasStatusEffect(StatusEffects.BLINDNESS)) {
+            toggleTime = System.currentTimeMillis();
+            isInitialized = true;
+        } else if (isInitialized && System.currentTimeMillis() - toggleTime >= OFFSET_DURATION) {
+            onLividColorFound(client, currentColor);
+            if (!originLividFound) {
+                String lividName = LIVID_TO_FORMATTING.entrySet().stream()
+                        .filter(entry -> entry.getValue() == color)
+                        .map(Map.Entry::getKey)
+                        .findFirst()
+                        .orElse("unknown");
+                client.world.getPlayers().stream()
+                        .filter(entity -> entity.getName().getString().equals(lividName))
+                        .findFirst()
+                        .ifPresent(entity -> lividID = entity.getId());
+                originLividFound = true;
```

Comment:
I'm a bit confused. How does this work when livid changes color when it only runs on the first time?

Actually, this whole `lividId` thing seems kinda unnecessary since we're still checking the name and you could revert to the old `names.contains` check if you end up changing this.

Reply:
The idea is to identify the original livid and use its ID to highlight the correct livid. The problem with the old variant is that the original Livid in M5 changes color, resulting in two Livids with the same color. Therefore, the only reliable identifier for the original Livid is its ID.

The code looks like that because i first need the livid name of the original livid and second go through all players in the area (in theory i could do 2 loops to get the info if you want it less confusing).

Reply:
I see, I wasn't familiar with M5 mechanics. Thanks for clearing this up.